src/compiler/checker.ts

```diff
@@ -11143,12 +11144,13 @@ namespace ts {
                     someHaveQuestionToken = someHaveQuestionToken || hasQuestionToken(node);
                     allHaveQuestionToken = allHaveQuestionToken && hasQuestionToken(node);
 
-                    if (nodeIsPresent(node.body) && bodyDeclaration) {
+                    if (nodeIsPresent(node.body) && implementationDeclaration) {
                         if (isConstructor) {
                             multipleConstructorImplementation = true;
                         }
                         else {
                             duplicateFunctionDeclaration = true;
+                            duplicateDefaultExports = !!(node.flags & NodeFlags.Default);

```

Comment:
No, these are all declarations of the same symbol, so if one is a default export, they should all be default exports.

Still, that's a great assertion to check here. I'll add it.